src/main/java/frc/robot/robot_manager/RobotManager.java

```diff
@@ -1,18 +1,207 @@
+import frc.robot.queuer.QueuerSubsystem;
+import frc.robot.shooter.ShooterState;
+import frc.robot.shooter.ShooterSubsystem;
 import frc.robot.util.scheduling.SubsystemPriority;
 import frc.robot.util.state_machines.StateMachine;
+import frc.robot.vision.VisionSubsystem;
 
 public class RobotManager extends StateMachine<RobotState> {
-  public RobotManager() {
+  public final ArmSubsystem arm;
+  public final ShooterSubsystem shooter;
+  public final LocalizationSubsystem localization;
+  public final VisionSubsystem vision;
+  public final ImuSubsystem imu;
+  public final IntakeSubsystem intake;
+  public final QueuerSubsystem queuer;
+
+  private final double distanceToFeedSpot = 0.0;
+  private final double distanceToSpeaker = 0.0;
+
+  public RobotManager(
+      ArmSubsystem arm,
+      ShooterSubsystem shooter,
+      LocalizationSubsystem localization,
+      VisionSubsystem vision,
+      ImuSubsystem imu,
+      IntakeSubsystem intake,
+      QueuerSubsystem queuer) {
     super(SubsystemPriority.ROBOT_MANAGER, RobotState.IDLE_NO_GP);
+    this.arm = arm;
+    this.shooter = shooter;
+    this.localization = localization;
+    this.vision = vision;
+    this.imu = imu;
+    this.intake = intake;
+    this.queuer = queuer;
   }
 
   @Override
   protected void collectInputs() {}
 
+  @Override
+  protected RobotState getNextState(RobotState currentState) {
+    switch (currentState) {
+      case SPEAKER_WAITING,
+          AMP_WAITING,
+          FEEDING_WAITING,
+          PASS_WAITING,
+          IDLE_NO_GP,
+          IDLE_WITH_GP,
+          CLIMBING_1_LINEUP,
+          CLIMBING_2_HANGING -> {}
+      case SPEAKER_SCORING -> {
+        if (!queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+      case AMP_SCORING -> {
+        if (!queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+      case FEEDING_SHOOTING -> {
+        if (!queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+      case PASS_SHOOTING -> {
+        if (!queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+      case SPEAKER_PREPARE_TO_SCORE -> {
+        if (shooter.atGoal() && arm.atGoal()) {
+          setStateFromRequest(RobotState.SPEAKER_SCORING);
+        }
+      }
+
+      case AMP_PREPARE_TO_SCORE -> {
+        if (shooter.atGoal() && arm.atGoal()) {
+          setStateFromRequest(RobotState.AMP_SCORING);
+        }
+      }
+      case FEEDING_PREPARE_TO_SHOOT -> {
+        if (shooter.atGoal() && arm.atGoal()) {
+          setStateFromRequest(RobotState.FEEDING_SHOOTING);
+        }
+      }
+      case PASS_PREPARE_TO_SHOOT -> {
+        if (shooter.atGoal() && arm.atGoal()) {
+          setStateFromRequest(RobotState.PASS_SHOOTING);
+        }
+      }
+      case UNJAM -> {}
+      case INTAKING -> {
+        if (intake.hasNote() || queuer.hasNote()) {
```

Comment:
for this i think we only want to check the queuer sensor, since that's the inner-most sensor (and should only trigger once the note is fully intaked)